internal/cf/resourcecache.go

```diff
@@ -0,0 +1,386 @@
+/*
+SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and cf-service-operator contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cf
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/sap/cf-service-operator/internal/facade"
+)
+
+// The resource cache is a simple in-memory cache to store CF resources like spaces, instances and
+// bindings using a map protected by a mutex.
+// The resource cache is used to avoid making multiple calls to the CF API and avoid rate limits.
+// The map uses the owner of the instance (which is Kubernetes UID) as key and the service instance
+// as value.
+type resourceCache struct {
+	mutex sync.RWMutex
+
+	// TODO: document which keys are used for the maps
+	// TODO: check if the keys for resources like spaces are unique across all CF organziations
+	// Scenario:
```

Comment:
The keys used for the maps are the label owner which is constructed this way, "service-operator.cf.cs.sap.com/owner": "K8s UID".
This applies to the bindings, spaces, and instances. Not sure if that answers these two TODOs.